src/js/libcs/CSNumber.js

```diff
@@ -765,180 +765,14 @@ CSNumber._helper.solveCubicHelper = function(a, b, c, d) {
     ]);
 };
 
+CSNumber.getRandReal = function(min, max) {
+    var mmin = min || 0;
+    var mmax = max || 1;
+    return CSNumber.real(Math.random() * (mmax - mmin) + mmin);
+};
```

Comment:
I'm not completely happy with the handling of default parameters here. Intuitively I'd expect `getRandReal(7)` to return something from the range [0, 7], or perhaps [-7, 7], but not [1, 7] as it does. Perhaps more serious: I'd expect `getRandReal(-1, 0)` to return a value from the range [-1, 0], but since zero is falsey, I'd actually get a result from the `[-1, 1]` range.

I know none of this affects how you are *currently* using the function. But it might still be confusing to future developers. So I suggest you either drop the substitution of default values (and force developers to always provide two arguments or to implement defaults *when the need arises*), or you do something like this:

```js
    var mmin, mmax;
    if (max !== undefined) {
        mmin = min;
        mmax = max;
    } else if (min !== undefined) {
        mmin = -min; // symmetric by default, or do we prefer zero here?
        mmax = min;
    } else {
        mmin = -1; // symmetric as well, for the sake of consistency
        mmax = 1;
    }
```

The same holds for the other random generating functions, although if you decide to allow default arguments, then in many cases you can just pass arguments along without bothering which of them are defined and which are not, letting the `getRandReal` (and perhaps also the `getRandComplex`) function handle the case distinctions.